service/matching/version_rule_helpers.go

```diff
@@ -489,12 +497,24 @@ func given2ActualIdx(idx int32, rules []*persistencespb.AssignmentRule) int {
 	return -1
 }
 
-// validRamp returns true if the percentage ramp is within [0, 100), or if the ramp is nil
-func validRamp(ramp *taskqueue.RampByPercentage) bool {
-	if ramp == nil {
-		return true
+// handleAssignmentRuleNilRamp converts a rule with nil ramp to have 100% ramp and returns the same rule object
+func handleAssignmentRuleNilRamp(rule *taskqueue.BuildIdAssignmentRule) *taskqueue.BuildIdAssignmentRule {
+	if rule.GetRamp() == nil {
+		rule.Ramp = &taskqueue.BuildIdAssignmentRule_PercentageRamp{
+			PercentageRamp: &taskqueue.RampByPercentage{RampPercentage: 100},
+		}
 	}
-	return ramp.RampPercentage >= 0 && ramp.RampPercentage < 100
+	return rule
+}
+
+// isValidRamp returns true if the percentage ramp is within [0, 100]
+func isValidRamp(ramp *taskqueue.RampByPercentage) bool {
+	return ramp.RampPercentage >= 0 && ramp.RampPercentage <= 100
+}
+
+// isPartiallyRamped returns true if the rule has a ramp percentage < 100
+func isPartiallyRamped(rule *taskqueue.BuildIdAssignmentRule) bool {
+	return rule.GetPercentageRamp().GetRampPercentage() < 100
```

Comment:
We should still handle nil ramp as 100%. There may be rules with nil ramp in the DB from past.

Reply:
ahh you're right. my bad

Reply:
fixed and tests to test it